examples/cosmos/ics20/ics20.qnt

```diff
@@ -325,6 +350,25 @@ module ics20Test {
     chainStates' = chainStates.set(chain, newChainState)
   }
 
+  /// Receive a timeout on `chain`:
+  /// - Pick an arbitrary, previously unreceived timeout on `chain`.
+  /// - Call the `onTimeoutPacket` callback on it.
+  /// - Update the chain state to record the timeout as received.
+  action receiveTimeout(chain: str): bool = all {
+    // (1) Compute updated chain state:
+    // non-deterministically pick an acknowledgement
+    val chainState = chainStates.get(chain)
+    nondet timedoutPacket = chainState.inTimeouts.oneOf()
```

Comment:
`oneOf` requires the set to be non-empty. So you have to add a test for it before calling `oneOf`. It's a bit inconvenient.

Reply:
Yes, we have to do this everywhere. I wanted to introduce it separately, together with a `step` action. Currently it does not matter much, because when writing tests, we know when the action can succeed (i.e., when the underlying set is non-empty).

Reply:
Fixed in 4530c46